### DISCUSSION TITLE
Undirected Probe Request Problem

### DISCUSSION THREAD

**Discussion by Jake-Grafton** [ZerBea/hcxdumptool#389] [Q&A] [answered]
Hello! When using **hcxdumptool** to dump information from channels that are used around my location, and then using **hcxpcapngtool** to convert the raw .pcapng files to a hashcat readable format, I get the error - 
Information: missing frames!
This dump file does not contain undirected proberequest frames.
An undirected proberequest may contain information about the PSK.
It always happens if the capture file was cleaned or
it could happen if filter options are used during capturing.
That makes it hard to recover the PSK.
The only filter options I use are telling **hcxdumptool** to hop on specific channels.
Here is the command that I use: **hcxdumptool -i wlan0 -c 1a,4a,6a,7a,9a,10a,11a,36b,40b,44b,48b,149b,153b,157b,161b --bpf=/home/jake/filters/protect.bpf --rds=1 -w test.pcapng**
The BPF just protects my home network and unwanted networks from being attacked.
Any help is appreciated!

**Comment by ZerBea** [comment]
2 different kinds of PROBEREQUEST frames:
directed:
`to MAC AP from MAC CLIENT and SSID (ESSID) is set`

and undirected:
```
to MAC BRAODCAST (ffffffffffff) from MAC CLIENT and wildcard SSID (ESSID)
to MAC BRAODCAST (ffffffffffff) from MAC CLIENT and SSID (ESSID) is set
```
This frames are very interesting, because the can contain some useful information.

To allow this channels "36b,40b,44b,48b,149b,153b,157b,161b" it is mandatory to set a wireless regulatory domain which allows this.

**Reply by Jake-Grafton** [reply]
Using **hcxdumptool -m wlan0** to put my adapter into monitor gives me this -

Requesting physical interface capabilities. This may take some time.
Please be patient...

interface information:

phy idx hw-mac       virtual-mac  m ifname           driver (protocol)
---------------------------------------------------------------------------------------------
  0   3 4822543041b5 4822543041b5 + wlan0            rtw_8822bu (NETLINK)


available frequencies: frequency [channel] tx-power of Regulatory Domain: 00

  2412 [  1] 20.0 dBm	  2417 [  2] 20.0 dBm	  2422 [  3] 20.0 dBm	  2427 [  4] 20.0 dBm
  2432 [  5] 20.0 dBm	  2437 [  6] 20.0 dBm	  2442 [  7] 20.0 dBm	  2447 [  8] 20.0 dBm
  2452 [  9] 20.0 dBm	  2457 [ 10] 20.0 dBm	  2462 [ 11] 20.0 dBm	  2467 [ 12] 20.0 dBm
  2472 [ 13] 20.0 dBm	  2484 [ 14] 20.0 dBm	  5180 [ 36] 20.0 dBm	  5200 [ 40] 20.0 dBm
  5220 [ 44] 20.0 dBm	  5240 [ 48] 20.0 dBm	  5260 [ 52] 20.0 dBm	  5280 [ 56] 20.0 dBm
  5300 [ 60] 20.0 dBm	  5320 [ 64] 20.0 dBm	  5500 [100] 20.0 dBm	  5520 [104] 20.0 dBm
  5540 [108] 20.0 dBm	  5560 [112] 20.0 dBm	  5580 [116] 20.0 dBm	  5600 [120] 20.0 dBm
  5620 [124] 20.0 dBm	  5640 [128] 20.0 dBm	  5660 [132] 20.0 dBm	  5680 [136] 20.0 dBm
  5700 [140] 20.0 dBm	  5720 [144] 20.0 dBm	  5745 [149] 20.0 dBm	  5765 [153] 20.0 dBm
  5785 [157] 20.0 dBm	  5805 [161] 20.0 dBm	  5825 [165] 20.0 dBm


monitor mode is active...
This output indicates that the regulatory domain allows the use of "36b,40b,44b,48b,149b,153b,157b,161b" and many more frequencies.

**Comment by ZerBea** [comment, marked as answer]
No, the regulatory domain is unset (00: DFS-UNSET):
`available frequencies: frequency [channel] tx-power of Regulatory Domain: 00`

```
$ iw reg get
global
country 00: DFS-UNSET
	(755 - 928 @ 2), (N/A, 20), (N/A), PASSIVE-SCAN
	(2402 - 2472 @ 40), (N/A, 20), (N/A)
	(2457 - 2482 @ 20), (N/A, 20), (N/A), AUTO-BW, PASSIVE-SCAN
	(2474 - 2494 @ 20), (N/A, 20), (N/A), NO-OFDM, PASSIVE-SCAN
	(5170 - 5250 @ 80), (N/A, 20), (N/A), AUTO-BW, PASSIVE-SCAN
	(5250 - 5330 @ 80), (N/A, 20), (0 ms), DFS, AUTO-BW, PASSIVE-SCAN
	(5490 - 5730 @ 160), (N/A, 20), (0 ms), DFS, PASSIVE-SCAN
	(5735 - 5835 @ 80), (N/A, 20), (N/A), PASSIVE-SCAN
	(57240 - 63720 @ 2160), (N/A, 0), (N/A)
```

The settings do not allow hcxdumptool to run active scans.

Some examples:
```
$ sudo iw reg set IN
$ iw reg get
global
country IN: DFS-UNSET
	(2402 - 2482 @ 40), (N/A, 30), (N/A)
	(5150 - 5250 @ 80), (N/A, 30), (N/A)
	(5250 - 5350 @ 80), (N/A, 24), (0 ms), DFS
	(5470 - 5725 @ 160), (N/A, 24), (0 ms), DFS
	(5725 - 5875 @ 80), (N/A, 30), (N/A)

$ sudo iw reg set IN
$ iw reg get
global
country US: DFS-FCC
	(902 - 904 @ 2), (N/A, 30), (N/A)
	(904 - 920 @ 16), (N/A, 30), (N/A)
	(920 - 928 @ 8), (N/A, 30), (N/A)
	(2400 - 2472 @ 40), (N/A, 30), (N/A)
	(5150 - 5250 @ 80), (N/A, 23), (N/A), AUTO-BW
	(5250 - 5350 @ 80), (N/A, 24), (0 ms), DFS, AUTO-BW
	(5470 - 5730 @ 160), (N/A, 24), (0 ms), DFS
	(5730 - 5850 @ 80), (N/A, 30), (N/A), AUTO-BW
	(5850 - 5895 @ 40), (N/A, 27), (N/A), NO-OUTDOOR, AUTO-BW, PASSIVE-SCAN
	(5925 - 7125 @ 320), (N/A, 12), (N/A), NO-OUTDOOR, PASSIVE-SCAN
	(57240 - 71000 @ 2160), (N/A, 40), (N/A)
```

**Reply by Jake-Grafton** [reply]
Ah, I see. I've set my regulatory domain to my region and it's definitely helping. One last question, should I use **-F** and allow hcxdumptool to hop on all available frequencies, or should I set the channel list manually using **-c**?

**Comment by ZerBea** [comment]
That depend on the traffic that you expect - and of course the kind of the target(s). CLIENTs will come to hcxdumptool (regardless the channel or the band).
A rcascan will give an overview of the operating channels which are really in use.
Too many scan channels in combination with a short stay time on the channel will reduce the chance to get a handshake.

**Reply by Jake-Grafton** [reply]
Good to know. Thanks for the help ZerBea!

**Comment by ZerBea** [comment]
No problem, you're welcome.
